pkg/ccl/logictestccl/testdata/logic_test/as_of

```diff
@@ -29,6 +29,9 @@ BEGIN AS OF SYSTEM TIME follower_read_timestamp(); SELECT * FROM t
 statement ok
 ROLLBACK
 
+statement error pgcode 0A000 cannot specify AS OF SYSTEM TIME with different timestamps
```

Comment:
ah actually i just realized, this is affecting 22.1, which has `enable_implicit_transaction_for_batch_statements` set to false.

based on the stack trace in https://github.com/cockroachdb/cockroach/issues/82179 it looks like this happened when executing a prepared statement (there is `execPortal` in the stack trace)

also it shows that it's from this line: https://github.com/cockroachdb/cockroach/blob/5b78463ed2e7106a8477b63fa837564ad02bb510/pkg/sql/conn_executor_exec.go#L776

Reply:
oh that's the line you updated. ok. i'm fine with this change

Reply:
i think it's because we also use an implicit transaction for the extended protocol now (https://github.com/cockroachdb/cockroach/pull/76792), so a test for that  too would be nice.

Reply:
do you know how it'd be triggered using the extended protocol?

Reply:
yeah, i think  if you do something like:

```
Parse SELECT .. AOST '-1us`
Bind
Exec
Parse SELECT ... AOST '-4us'
Bind
Exec
Sync
```

hmm but that wouldn't cause it during ExecPortal

what if you do
```
Parse SELECT .. AOST follower_read_timestamp()
Bind
Exec
Sync
```

Reply:
dones